### PR TITLE
Fix Booleans at njsDbObject_transformFromOracle

### PR DESCRIPTION
**Pull request by ArcezD**
1. Support 'PL/SQL BOOLEAN'  at tranformation from DBObject to Javascript values.

Signed-off-by: Diego Arce <Diego@arce.cr>

#1189

### CODE REVIEW

**Comment by cjbj** [comment]
Can you point me at your [OCA entry](https://github.com/oracle/node-oracledb/blob/master/CONTRIBUTING.md)?

**Comment by ArcezD** [comment]
> Can you point me at your [OCA entry](https://github.com/oracle/node-oracledb/blob/master/CONTRIBUTING.md)?

I emailed it in few hours ago.

**Comment by cjbj** [comment]
That's great!  I'll ask the OCA team to keep an eye out for it.

File: src/njsVariable.c [hunk 1]
```diff
@@ -531,6 +542,10 @@ bool njsVariable_initForQuery(njsVariable *vars, uint32_t numVars,
             case DPI_ORACLE_TYPE_NATIVE_DOUBLE:
             case DPI_ORACLE_TYPE_ROWID:
                 break;
+            case DPI_ORACLE_TYPE_BOOLEAN:
+                vars[i].varTypeNum = DPI_ORACLE_TYPE_BOOLEAN;
+                vars[i].nativeTypeNum = DPI_NATIVE_TYPE_BOOLEAN;
+                break;
```
Comment (anthony-tuininga) [inline src/njsVariable.c]: This section is not needed as booleans cannot be fetched. They can only be used in PL/SQL.

File: src/njsVariable.c [hunk 2]
```diff
@@ -235,6 +239,8 @@ uint32_t njsVariable_getDataType(njsVariable *var)
             return NJS_DATATYPE_CLOB;
         case DPI_ORACLE_TYPE_BLOB:
             return NJS_DATATYPE_BLOB;
+        case DPI_ORACLE_TYPE_BOOLEAN:
+            return NJS_DATATYPE_BOOLEAN;
```
Comment (anthony-tuininga) [inline src/njsVariable.c]: This section is not needed as booleans cannot be fetched. They can only be used in PL/SQL.

**Comment by cjbj** [comment]
@ArcezD thanks for getting the OCA submitted.  You can see @anthony-tuininga has some comments; more on this below.

If you're not aware of the process we use, we'll merge your PR to a private git repo (where we work on things like not-yet released Oracle DB features), and then merge back to GitHub, and make a release.

For timing reasons, don't worry about making the trivial changes noted by @anthony-tuininga.  He'll make them when we merge the PR.  We'll add tests and doc too.

**Comment by cjbj** [comment]
@ArcezD this finally landed in [node-oracledb 4.2](https://blogs.oracle.com/opal/node-oracledb-42-with-improved-data-type-binding).  It coincided nicely with the other bind changes.  Thanks!

**Pull request closed without merging**